File: sdk/typescript/libraries/botbuilder-solutions/src/util/confirmRecognizerHelper.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { ModelResult } from '@microsoft/recognizers-text';
import { recognizeBoolean } from '@microsoft/recognizers-text-choice';
import { PromptRecognizerResult } from 'botbuilder-dialogs';

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace ConfirmRecognizerHelper {
    const valueKey = 'value';

    export function confirmYerOrNo(utterance: string, locale: string = ''): PromptRecognizerResult<boolean> {
        let result: PromptRecognizerResult<boolean> = { succeeded: false };

        if (utterance !== undefined && utterance.trim().length > 0) {
            // Recognize utterance
            const results: ModelResult[] = recognizeBoolean(utterance, locale);

            if (results.length > 0) {
                const first: ModelResult = results[0];
                const value: boolean = first.resolution[valueKey] as boolean;
                if (value !== undefined) {
                    result = {
                        succeeded: true,
                        value: value
                    };
                }
            }
        }

        return result;
    }
}

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Util/ConfirmRecognizerHelper.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using Microsoft.Bot.Builder.Dialogs;
using Microsoft.Recognizers.Text.Choice;

namespace Microsoft.Bot.Builder.Solutions.Util
{
    [Obsolete("This type is being deprecated. It's moved to the assembly Microsoft.Bot.Solutions. Please refer to https://aka.ms/botframework-solutions/releases/0_8", false)]
    public static class ConfirmRecognizerHelper
    {
        public static PromptRecognizerResult<bool> ConfirmYesOrNo(string utterance, string locale = null)
        {
            var result = new PromptRecognizerResult<bool>();
            if (!string.IsNullOrEmpty(utterance))
            {
                // Recognize utterance
                var results = ChoiceRecognizer.RecognizeBoolean(utterance, locale);
                if (results.Count > 0)
                {
                    var first = results[0];
                    if (bool.TryParse(first.Resolution["value"].ToString(), out var value))
                    {
                        result.Succeeded = true;
                        result.Value = value;
                    }
                }
            }

            return result;
        }
    }
}
